test/SIM_mc_generation/RUN_ERROR_invalid_call/input.py

```diff
@@ -15,7 +15,7 @@
                                        "Modified_data/datafile.txt",
                                        3)
 # the next command is the source of the error!
-mc_var.register_dependent(0)
+mc_var.register_dependent(None)
```

Comment:
Why pass none instead of 0?

Reply:
This one I need to dig into more on the "why". But if you pass in `0` the SWIG layer was throwing a bad argument exception.  After looking at the source code for `register_dependent`, it's clear that the intent is to check for `NULL`, and in `python` `None` is the closest thing to `NULL` in C++. It's kind of weird, python doesn't have pointers like C++ does. Anywho, I'm not quite sure why this pops up in your implementation here but I don't see it in the CML ramtares version. It may have something to do with the Trick version being different. Either way, `None` is objectively more correct than `0` here and changing to `None` makes the test return what's expected wrt return codes of `./S_main...`